core/src/test/scala/caliban/execution/ExecutionSpec.scala

```diff
@@ -450,6 +450,70 @@ object ExecutionSpec extends DefaultRunnableSpec {
 
         assertM(interpreter.flatMap(_.execute(query)).map(_.data.toString))(equalTo("""{"i":{"id":"ok"}}"""))
       },
+      testM("rename on a union child") {
+        sealed trait Union
+        object Union {
+          case class Child(field: String) extends Union
+        }
+        case class Obj(union: Union)
+        case class Query(test: Obj)
+
+        object Schemas {
+          implicit val schemaUnionChild: Schema[Any, Union.Child] = Schema.gen[Union.Child].rename("UnionChild")
+          implicit val schemaTestUnion: Schema[Any, Union]        = Schema.gen[Union]
+          implicit val schemaQuery: Schema[Any, Query]            = Schema.gen[Query]
+        }
+        import Schemas._
+
+        val interpreter = graphQL(RootResolver(Query(Obj(Union.Child("f"))))).interpreter
+        val query       = gqldoc("""
+             {
+               test {
+                 union {
+                   __typename
+                   ... on UnionChild {
+                     field
+                   }
+                 }
+               }
+             }""")
+
+        assertM(interpreter.flatMap(_.execute(query)).map(_.data.toString))(
+          equalTo("""{"test":{"union":{"__typename":"UnionChild","field":"f"}}}""".stripMargin)
+        )
+      },
+      testM("rename on a union child and parent") {
+        sealed trait Union
+        object Union {
+          case class Child(field: String) extends Union
+        }
+        case class Obj(union: Union)
+        case class Query(test: Obj)
+
+        object Schemas {
+          implicit val schemaUnionChild: Schema[Any, Union.Child] = Schema.gen[Union.Child].rename("UnionChild")
+          implicit val schemaTestUnion: Schema[Any, Union]        = Schema.gen[Union].rename("UnionRenamed")
+          implicit val schemaQuery: Schema[Any, Query]            = Schema.gen[Query]
+        }
+        import Schemas._
+
+        val interpreter = graphQL(RootResolver(Query(Obj(Union.Child("f"))))).interpreter
+        val query       = gqldoc("""
+             {
+               test {
+                 union {
+                   __typename
+                   ... on UnionChild {
+                     field
+                   }
+                 }
+               }
+             }""")
+
+        assertM(interpreter.flatMap(_.execute(query)).map(_.data.toString))(
+          equalTo("""{"test":{"union":{"__typename":"UnionChild","field":"f"}}}""".stripMargin)
+        )
```

Comment:
I think should also verify that the parent got renamed.

Reply:
The parent will never be returned by `__typename`, right?

Reply:
It would show up in the introspection / schema though I think.

Reply:
Right. This isn't affected by this PR though so I'll merge this one (finally 😄)